<a href="https://colab.research.google.com/github/MCecileD/Food_NLP/blob/main/Photos_contr%C3%B4le.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Import des bibliothèques nécessaires

In [2]:
import pandas as pd
import os
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


Lister les fichiers/photos du répertoire

In [4]:
dir_photos = "drive/MyDrive/Nao_School/week8/dataset/Data_Set_Projet_2/Food Images/Food Images"

In [5]:
!ls 'drive/MyDrive/Nao_School/week8/dataset/Data_Set_Projet_2/Food Images/Food Images' -1 | wc -l


461


Création d'une boucle pour vérifier le nom des fichiers présents dans le répertoire

In [6]:
list_photos =[]
for file in os.listdir(f"{dir_photos}"):
  list_photos.append(file)

In [16]:
list_photos

['-em-ba-em-s-ultimate-lobster-rolls-51169080.jpg',
 'almond-cake-366229.jpg',
 'anchovy-and-rosemary-roasted-lamb-231791.jpg',
 'all-green-salad-with-citrus-vinaigrette-56389342.jpg',
 '3-ingredient-pesto-fried-chicken.jpg',
 '4-3-2-1-spice-rub-51175040.jpg',
 '3-ingredient-red-snapper-with-green-sauce-and-scallions.jpg',
 '3-ingredient-smoked-mozzarella-sticks.jpg',
 'angry-shrimp-51219630.jpg',
 'apple-galettes-with-caramel-sauce-240489.jpg',
 'apricot-anise-tarts-395030.jpg',
 'apple-walnut-ginger-galette-368305.jpg',
 'apple-tart-with-caramel-sauce-236507.jpg',
 'apple-pie-cookies.jpg',
 'barbecued-chicken-and-chickpea-quesadillas-363314.jpg',
 'bratwurst-and-red-cabbage-51263820.jpg',
 'black-rice-pudding-233288.jpg',
 'baked-chicken-with-white-beans-and-tomatoes-231182.jpg',
 'bazooka-bubblegum-cocktail-350146.jpg',
 'basic-butter-cookie-dough-13504.jpg',
 'brisket-with-pomegranate-walnut-sauce-and-pistachio-gremolata.jpg',
 'breakfast-polenta-with-chorizo-and-queso-fresco-23127

Création dataframe avec les noms des photos

In [7]:
df =  pd.DataFrame(columns=['Nom_entier','Nom_sans_format'])

In [8]:
df

Empty DataFrame
Columns: [Nom_entier, Nom_sans_format]
Index: []

In [10]:
df['Nom_entier'] = list_photos 

Formatter le nom des fichiers pour la comparaison

In [11]:
df['Nom_sans_format'] = df['Nom_entier'].str.slice(0,-4)

In [12]:
df

Nom_entier  \
0      -em-ba-em-s-ultimate-lobster-rolls-51169080.jpg   
1                               almond-cake-366229.jpg   
2         anchovy-and-rosemary-roasted-lamb-231791.jpg   
3    all-green-salad-with-citrus-vinaigrette-563893...   
4                 3-ingredient-pesto-fried-chicken.jpg   
..                                                 ...   
456                  tomato-coconut-curry-with-cod.jpg   
457                                   tonatto-eggs.jpg   
458               vanilla-bean-rice-pudding-351549.jpg   
459     watercress-orange-and-avocado-salad-233390.jpg   
460      wilted-spinach-with-roasted-garlic-240366.jpg   

                                      Nom_sans_format  
0         -em-ba-em-s-ultimate-lobster-rolls-51169080  
1                                  almond-cake-366229  
2            anchovy-and-rosemary-roasted-lamb-231791  
3    all-green-salad-with-citrus-vinaigrette-56389342  
4                    3-ingredient-pesto-fried-chicken  
..                                                ...  
456                     tomato-coconut-curry-with-cod  
457                                      tonatto-eggs  
458                  vanilla-bean-rice-pudding-351549  
459        watercress-orange-and-avocado-salad-233390  
460         wilted-spinach-with-roasted-garlic-240366  

[461 rows x 2 columns]

Répertoire du CSV

In [13]:
dir_csv = "drive/MyDrive/Nao_School/week8/dataset/Data_Set_Projet_2/"

récupération du Food.csv

In [14]:
df_csv = pd.read_csv(dir_csv + '/Food.csv', delimiter=",")

In [43]:
df_csv

Unnamed: 0                                              Title  \
0               0  Miso-Butter Roast Chicken With Acorn Squash Pa...   
1               1                    Crispy Salt and Pepper Potatoes   
2               2                        Thanksgiving Mac and Cheese   
3               3                 Italian Sausage and Bread Stuffing   
4               4                                       Newton's Law   
...           ...                                                ...   
13496       13496                               Brownie Pudding Cake   
13497       13497  Israeli Couscous with Roasted Butternut Squash...   
13498       13498  Rice with Soy-Glazed Bonito Flakes and Sesame ...   
13499       13499                                        Spanakopita   
13500       13500  Mexican Poblano, Spinach, and Black Bean "Lasa...   

                                             Ingredients  \
0      ['1 (3½–4-lb.) whole chicken', '2¾ tsp. kosher...   
1      ['2 large egg whites', '1 pound new potatoes (...   
2      ['1 cup evaporated milk', '1 cup whole milk', ...   
3      ['1 (¾- to 1-pound) round Italian loaf, cut in...   
4      ['1 teaspoon dark brown sugar', '1 teaspoon ho...   
...                                                  ...   
13496  ['1 cup all-purpose flour', '2/3 cup unsweeten...   
13497  ['1 preserved lemon', '1 1/2 pound butternut s...   
13498  ['Leftover katsuo bushi (dried bonito flakes) ...   
13499  ['1 stick (1/2 cup) plus 1 tablespoon unsalted...   
13500  ['12 medium to large fresh poblano chiles (2 1...   

                                            Instructions  \
0      Pat chicken dry with paper towels, season all ...   
1      Preheat oven to 400°F and line a rimmed baking...   
2      Place a rack in middle of oven; preheat to 400...   
3      Preheat oven to 350°F with rack in middle. Gen...   
4      Stir together brown sugar and hot water in a c...   
...                                                  ...   
13496  Preheat the oven to 350°F. Into a bowl sift to...   
13497  Preheat oven to 475°F.\nHalve lemons and scoop...   
13498  If using katsuo bushi flakes from package, moi...   
13499  Melt 1 tablespoon butter in a 12-inch heavy sk...   
13500  Lay 4 chiles on their sides on racks of gas bu...   

                                              Image_Name  \
0      miso-butter-roast-chicken-acorn-squash-panzanella   
1             crispy-salt-and-pepper-potatoes-dan-kluger   
2             thanksgiving-mac-and-cheese-erick-williams   
3              italian-sausage-and-bread-stuffing-240559   
4                     newtons-law-apple-bourbon-cocktail   
...                                                  ...   
13496                         brownie-pudding-cake-14408   
13497  israeli-couscous-with-roasted-butternut-squash...   
13498  rice-with-soy-glazed-bonito-flakes-and-sesame-...   
13499                                 spanakopita-107344   
13500  mexican-poblano-spinach-and-black-bean-lasagne...   

                                     Cleaned_Ingredients  
0      ['1 (3½–4-lb.) whole chicken', '2¾ tsp. kosher...  
1      ['2 large egg whites', '1 pound new potatoes (...  
2      ['1 cup evaporated milk', '1 cup whole milk', ...  
3      ['1 (¾- to 1-pound) round Italian loaf, cut in...  
4      ['1 teaspoon dark brown sugar', '1 teaspoon ho...  
...                                                  ...  
13496  ['1 cup all-purpose flour', '2/3 cup unsweeten...  
13497  ['1 preserved lemon', '1 1/2 pound butternut s...  
13498  ['Leftover katsuo bushi (dried bonito flakes) ...  
13499  ['1 stick (1/2 cup) plus 1 tablespoon unsalted...  
13500  ['12 medium to large fresh poblano chiles (2 1...  

[13501 rows x 6 columns]

Création d'une colonne de présence des photos dans le CSV, on devrait obtenir 13501 images à True

In [15]:
df["PrésentdansCSV"] = df["Nom_sans_format"].isin(df_csv["Image_Name"])


Compter le nbre de True

In [16]:
len(df[df["PrésentdansCSV"]==True])

454

In [17]:
len(df[df["PrésentdansCSV"]==False])

7

Créer un repertoire avec uniquement les photos présentes à True

In [86]:
# Filtrer les photos disponibles
photos = df[df["PrésentdansCSV"] == True]["Nom_entier"]

# Créer le répertoire pour stocker les photos
# os.system("mkdir drive/MyDrive/Nao_School/week8/dataset/Data_Set_Projet_2/photos")
new_path ="drive/MyDrive/Nao_School/week8/dataset/Data_Set_Projet_2/photos/"
# Copier les photos filtrées dans le répertoire
for photo in photos:
     !cp "drive/MyDrive/Nao_School/week8/dataset/Data_Set_Projet_2/Food Images/Food Images/$photo" "$new_path"


Vérification du CSV avec le répertoire, les Image_Name peuvent ne pas avoir de photos (contrôle caduque si on a le même nombre total de photos et recettes)

In [18]:
df_csv["Présentdansrepphotos"] = df_csv["Image_Name"].isin(df["Nom_sans_format"])

Compter le nbre de True => Image_Name dans le répertoire ?

In [19]:
len(df_csv[df_csv["Présentdansrepphotos"]==True])

454

In [20]:
len(df_csv[df_csv["Présentdansrepphotos"]==False])

13047

Vérification du nbre de photos dans le nouveau répertoire créé photos

In [87]:
!ls 'drive/MyDrive/Nao_School/week8/dataset/Data_Set_Projet_2/photos/' -1 | wc -l

454


Dernier contrôle des photos dans le nouveau répertoire avec le fichier CSV

In [91]:
new_list_photos =[]
for file in os.listdir(f"{new_path}"):
  new_list_photos.append(file)

In [92]:
df_contr =  pd.DataFrame(columns=['Nom_entier','Nom_sans_format'])

In [94]:
df_contr['Nom_entier'] = new_list_photos 

In [95]:
df_contr['Nom_sans_format'] = df_contr['Nom_entier'].str.slice(0,-4)

In [96]:
df_contr["PrésentdansCSV"] = df_contr["Nom_sans_format"].isin(df_csv["Image_Name"])

verification si aucun False

In [97]:
len(df_contr[df_contr["PrésentdansCSV"]==False])

0